### PR TITLE
[react-native-auth0] Add typing (Passwordless SMS)

### PR DESCRIPTION
**Pull request by yammesicka**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [here](https://github.com/auth0/react-native-auth0/blob/91c08f93d37e0dd2f2c41415ebb65f4cf720f1a4/src/auth/index.js#L264)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/react-native-auth0/index.d.ts [hunk 1]
```diff
@@ -109,6 +110,8 @@ export interface PasswordlessWithEmailParams {
 
 export interface PasswordlessWithSMSParams {
     phoneNumber: string;
+    send?: 'link' | 'code' | undefined;
```
Comment (ascariandrea) [inline types/react-native-auth0/index.d.ts:113]: The `undefined` shouldn't be needed as property is optional (`?`)

File: types/react-native-auth0/react-native-auth0-tests.ts [hunk 2]
```diff
@@ -158,6 +158,8 @@ auth0.auth.passwordlessWithEmail({
 
 auth0.auth.passwordlessWithSMS({
     phoneNumber: '+5491159991000',
+    send: 'code',
+    authParams: {scope: 'openid offline_access'},
```
Comment (ascariandrea) [inline types/react-native-auth0/react-native-auth0-tests.ts:162]: Did lint run for this file?

**Review by ascariandrea — APPROVED** [review]
Looks good to me now! 🥂

**Comment by yammesicka** [comment]
Ready to merge

File: types/react-native-auth0/index.d.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@
 //                 Leo Farias <https://github.com/leoafarias>
 //                 Will Dady <https://github.com/willdady>
 //                 Bogdan Vitoc <https://github.com/bogidon>
+//                 Yam Mesicka <https://github.com/yammesicka>
```
Comment (noyfactor) [inline types/react-native-auth0/index.d.ts:8]: 😍

**Pull request merged**